core/src/main/java/feast/core/controller/CoreServiceRestController.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.controller;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Timestamp;
+import feast.core.config.FeastProperties;
+import feast.core.model.Project;
+import feast.core.service.AccessManagementService;
+import feast.core.service.JobService;
+import feast.core.service.SpecService;
+import feast.core.service.StatsService;
+import feast.proto.core.CoreServiceProto.GetFeastCoreVersionResponse;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest.Builder;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsResponse;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsRequest;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsResponse;
+import feast.proto.core.CoreServiceProto.ListFeaturesRequest;
+import feast.proto.core.CoreServiceProto.ListFeaturesResponse;
+import feast.proto.core.CoreServiceProto.ListProjectsResponse;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * EXPERIMENTAL: Controller for HTTP endpoints to Feast Core. These endpoints are subject to change.
+ */
+@RestController
+@CrossOrigin
+@Slf4j
+@RequestMapping(value = "/api/v1", produces = "application/json")
+public class CoreServiceRestController {
+
+  private final FeastProperties feastProperties;
+  private SpecService specService;
+  private JobService jobService;
+  private StatsService statsService;
+  private AccessManagementService accessManagementService;
```

Comment:
Apologies for throwing code water at the last minute, and not in scope for this PR.
Is it possible for the CoreServiceRestController to call CoreServiceImpl?
This would remove much of the code duplication required to achieve feature parity.